File: src/types/assembly.ts
// Type definitions for National Assembly API
// Endpoint: https://www.parliament.bg/api/v1/fn-assembly/bg

export interface Assembly {
  A_ns_C_id: number;
  A_ns_CL_value: string;
  A_ns_CL_value_short: string;
  A_ns_C_active_count: number;
  A_ns_C_date_F: string; // Format: YYYY-MM-DD
  A_ns_C_date_T: string; // Format: YYYY-MM-DD (usually 9999-12-31)
}

export interface AssemblyResponse {
  A_ns_C_id: number;
  A_ns_CL_value: string;
  A_ns_CL_value_short: string;
  A_ns_C_active_count: number;
  colListMP?: unknown[]; // Not used in current implementation
}

File: backend/Models/Assembly.cs
using System.Text.Json.Serialization;

namespace ParliamentAbsenceVisualizer.Api.Models;

public class Assembly
{
    [JsonPropertyName("A_ns_C_id")]
    public int A_ns_C_id { get; set; }

    [JsonPropertyName("A_ns_CL_value")]
    public string A_ns_CL_value { get; set; } = string.Empty;

    [JsonPropertyName("A_ns_CL_value_short")]
    public string A_ns_CL_value_short { get; set; } = string.Empty;

    [JsonPropertyName("A_ns_C_active_count")]
    public int A_ns_C_active_count { get; set; }

    [JsonPropertyName("A_ns_C_date_F")]
    public string A_ns_C_date_F { get; set; } = string.Empty;

    [JsonPropertyName("A_ns_C_date_T")]
    public string A_ns_C_date_T { get; set; } = string.Empty;
}
